### PR TITLE
NP-3021: separate IndexingClient from SearchClient

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: search-commons/src/main/java/no/unit/nva/search/IndexingClient.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.support.ActiveShardCount;
+import org.elasticsearch.action.support.WriteRequest.RefreshPolicy;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestClientBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IndexingClient {
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/IndexingClient.java:41]: Code was cut-pasted from ElasticSearchHighLevelRestClient

File: search-commons/src/test/java/no/unit/nva/search/ElasticsearchSigningHighLevelRestClientTest.java [hunk 2]
```diff
@@ -111,90 +83,6 @@ void searchSingleTermReturnsErrorResponseWhenExceptionInDoSearch() throws IOExce
                                                                                            SortOrder.DESC));
     }
 
-    @Test
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/ElasticsearchSigningHighLevelRestClientTest.java:114]: tests were moved to IndexingClientTest class

File: search-commons/src/test/java/no/unit/nva/search/IndexingClientTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+import org.elasticsearch.client.RequestOptions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+
+public class IndexingClientTest {
+
+    public static final int NUMBER_NOT_DIVIDABLE_BY_BLOCK_SIZE = 1256;
+
+    @Test
+    void shouldCreateDefaultObjectWithoutFailing() {
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/IndexingClientTest.java:46]: This is a test for covering the default constructor

File: search-commons/src/test/java/no/unit/nva/search/IndexingClientTest.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        assertDoesNotThrow((Executable) IndexingClient::new);
+    }
+
+    @Test
+    void batchInsertIndexesAllDocumentsInBatchInBulksOfSpecifiedSize() throws IOException {
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/IndexingClientTest.java:51]: All tests were moved to from the ElasticSearchHighLevelRestClient and the only difference is the name of the client, which now is "IndexingClient"

**Review by brinxmat — APPROVED** [review]

**Pull request merged**